File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cs
using System;

public class Solution {
    public int solution(string myString, string pat) {
        int answer = 0;
        string temp = myString.Replace('A','@');
        temp = temp.Replace('B','A');
        temp = temp.Replace('@','B');
        answer = temp.Contains(pat) == true ? 1 : 0;
        return answer;
    }
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(string myString, string pat) {
    string answer = myString;
    replace(answer.begin(), answer.end(), 'A','C');
    replace(answer.begin(), answer.end(), 'B','A');
    replace(answer.begin(), answer.end(), 'C','B');
    return (answer.find(pat) != string::npos) ? 1 : 0;
}